src/plan/transitive_closure.rs

```diff
@@ -71,3 +71,163 @@ impl<'a, E: ProcessEdgesWork> Drop for ObjectsClosure<'a, E> {
         self.flush();
     }
 }
+
```

Comment:
It makes sense to move `PlanProcessEdges` to `scheduler/gc_work.rs`. What do you suggest about `PlanTraceObject`?

Splitting the `scheduler::gc_work` module should be a separate PR. I would prefer not doing it in this PR.

Reply:
@qinsoon `PlanTraceObject` can be in the same place as `trait Plan` itself, i.e. `src/plan/global.rs`.  It represents something that a `Plan` does, although mainly used by the `PlanProcessEdges` work packet (but it can be used elsewhere as well).  I suppose other methods, such as `get_used_pages`, can also be derived.

Reply:
I have moved `PlanTraceObject` to `plan/global.rs`, and moved `PlanProcessEdges` to `scheduler/gc_work.rs`.